docs/DeveloperGuide.md

```diff
@@ -1,38 +1,335 @@
+
+![](images/WithdrawCommandSeqDiagram.png)
+
+### View balance feature
+
+The view balance feature is facilitated using `Account` instances stored within the `AccountList` object. The main
+functionality is to view the balance of a specific currency if the currency is specified, else view all the currencies
+in the account.
+
+![BalanceSequenceDiagram](images/BalanceSeqDiagram.png)
+
+### Show-rate feature
+
+The exchange feature is facilitated using two `Forex` instances to represent the
+exchange rates between two currencies. The current implementation reads manual exchange
+rates from an online source. Future implementation will use an API to maintain up-to-date
+exchange rates.
+
+Exchange rate source: https://www.xe.com/currencyconverter/convert
+
+The show-rate command executes as follows
+
+- The initial and target currency are parsed from the input
+- If the user included an amount, the amount is also parsed
+- A `Forex` object showing the rate from the initial to the target currency is made
+- A `Forex` object showing the rate from the target to the initial currency is made
+- The Ui prints the exchanged amount if an amount was provided by the user for both rates
+- The Ui will print the unit rate both ways if no amount was provided
+
+The following sequence diagram shows how the Show Rate command works
+![ShowRateSeqDiagram](images/ShowRateSeqDiagram.png)
+
+### Money exchange feature
+
+The exchange feature is facilitated using `Account` instances stored within an `AccountList`
+object. The main functionality is facilitated by the `convert` function within the `Forex`
+component. The current implementation reads manual exchange rates from an online source. Future
+implementation will use an API to maintain up-to-date exchange rates. An exchange call will produce two
+transaction activity. A debit transaction will reflect on initial account and credit transaction will
+reflect on target account
+
+Exchange rate source: https://www.xe.com/currencyconverter/convert
+
+This command is executed under the assumption that an `Account` for both the initial and target
+currencies exist. To avoid redundancy, please see the `create-account` feature in the developer
+guide for more specific steps on how `Accounts` are created.
+
+The exchange command executes as follows:
+
+- Initial and target currencies are parsed from the user input
+- A Forex object is created using the parsed currencies (see `Forex` component for more information)
+- The amount to be exchanged is parsed from the user input
+- The `Accounts` for both currencies are retrieved
+- The converted value is calculated using the `Forex` object
+- The value of the initial `Account` is updated
+- The value of the target `Account` is updated
+- The new balances are printed
 
-## Design & implementation
+The following sequence diagram shows how the Exchange command works
+![ExchangeSeqDiagram](images/ExchangeSeqDiagram.png)
```

Comment:
Very detailed sequence diagram, but one point: is the spelling wrong? (underlined in red : BigDecimal instead of BigDecinmal)
![image](https://user-images.githubusercontent.com/88625158/228435043-3e2bb5b6-beed-42ab-8c32-f14dddb53142.png)